Source/WebCore/Modules/permissions/PermissionController.h

```diff
@@ -38,7 +38,7 @@ struct PermissionDescriptor;
 class PermissionController : public ThreadSafeRefCounted<PermissionController> {
 public:
     virtual ~PermissionController() = default;
-    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&) = 0;
+    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```

Comment:
Apple and WebKit style put callbacks like CompletionHander in the last argument position. This makes reading the call sites easier, since you see all the arguments before the multi-line block of callback code.

So, you should put ScriptExecutionContext before CompletionHandler here.
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```
```suggestion
    virtual void query(WebCore::ClientOrigin&&, PermissionDescriptor&&, CompletionHandler<void(std::optional<PermissionState>)>&&, ScriptExecutionContext*) = 0;
```

Reply:
I see what you mean about the style here. Thank you for the feedback! The implementation for this pull request has now changed, but I'll keep this in mind for the future.